python/tk_multi_loader/model_latestpublish.py

```diff
@@ -79,107 +79,129 @@ def load_data(self, item, child_folders, show_sub_items, additional_sg_filters):
-                elif entity_type == "Version":
-                    sg_filters = [["version", "in", data]]
-                else:
-                    sg_filters = [["entity", "in", data]]
-
-                # lastly, when we are in this special mode, the main view
-                # is no longer functioning as a browsable hierarchy
-                # but is switching into more of a paradigm of an inverse
-                # database. Indicate the difference by not showing any folders
-                child_folders = []
+                # loop through the selected items
+                for item in items:
+                    # special mode -- in this case we don't show any of the
+                    # child folders and only the partial matches of all the leaf nodes
+
+                    # for example, this may return
+                    # entity type shot, [["sequence", "is", "xxx"]] or
+                    # entity type shot, [["status", "is", "ip"]] or
+
+                    # note! Because of nasty bug https://bugreports.qt-project.org/browse/PYSIDE-158,
+                    # we cannot pull the model directly from the item but have to pull it from
+                    # the model index instead.
+                    model_idx = item.index()
+                    model = model_idx.model()
+                    partial_filters = model.get_filters(item)
+                    entity_type = model.get_entity_type()
+
+                    # now get a list of matches from the above query from
+                    # shotgun - note that this is a synchronous call so
+                    # it may 'pause' execution briefly for the user
+                    data = self.app.shotgun.find(entity_type, partial_filters)
+
+                    # now create the final query for the model - this will be
+                    # a big in statement listing all the ids returned from
+                    # the previous query, asking the model to only show the
+                    # items matching the previous query.
+                    #
+                    # note that for tasks, we link via the task field
+                    # rather than the std entity link field
+                    #
+
+                    # New sg_filter for tags. We need to pull the tag applied
+                    # to the Version associated with the publish
+                    # In the context of a media library it should be assumed
+                    # that any PublishedFile WILL have a Version associated with it.
+                    # We may need to add logic to cover cases where the published file has no version.
+                    if entity_type == "Task":
+                        sg_filters = [["task", "in", data]]
+                    elif entity_type == "Version":
+                        sg_filters = [["version", "in", data]]
+                    elif entity_type == "Tag":
+                        sg_filters = [["version.Version.tags", "in", data ]]
```

Comment:
I'm wondering if we want to expose this field in the config settings instead. for example, could i configure the tag matching for tags on the publish's linked entity instead? what do you think?

Reply:
True, I don't see why not. 
My use-case is specific; a media library that might be browsed via the loader2 or the website Media pages. As there is no way to filter the website 'Media' views by PublishedFile tag (eg, when you are using the Media view to browse a media library), the the tag must be on the Version.

This is also a structural limitation; there can be multiple published_files with different tags per Version, whereas there can only be one Version per multiple published files. If the tags are on the published files, there is no way to filter a list of Versions by any publishedFile fields is there?

It therefore, in my view, makes no sense to tag published files as the functionality would not extend beyond the workfiles app. At least, certainly from the use-case of a media library. Which is the key purpose of introducing support for Tag entities. 

Other use-cases might indeed benefit, eg where you don't have a Version associated with a published file, or where you need to tag publishes of a single Version with different tags....

Anyway, I'm all for leaving things as configurable as possible, so we should just do what you suggest. Can you tell me how to do that?